unity-client/Assets/Scripts/MainScripts/DCL/Components/Textures/WrappedTextureAsset/IPromiseLike_TextureAsset.cs

```diff
@@ -0,0 +1,8 @@
+namespace DCL
+{
+    public interface IPromiseLike_TextureAsset
```

Comment:
Why can't we just use the `AssetPromise_Texture` ?

Reply:
I needed a class that contains gifs or textures to avoid writing code to handle those cases separately... due to the generics contrains in `AssetPromise<AssetType>` and `AssetPromiseKeeper<AssetType, AssetLibraryType, AssetPromiseType>` I couldn't use the `ITexture` interface nor any other interface to achieve that, so a wrapper class was the only solution I could find